src/Type/MediaItem/MediaItemDelete.php

```diff
@@ -0,0 +1,136 @@
+					// translators: The placeholder is the name of the post's post_type being deleted
+					'description' => __( 'The ID of the mediaItem to delete', 'wp-graphql' ),
+				],
+				'forceDelete' => [
+					'type'        => Types::boolean(),
+					'description' => __( 'Whether the mediaItem should be force deleted instead of being moved to the trash', 'wp-graphql' ),
+				],
+			],
+			'outputFields'        => [
+				'deletedId' => [
+					'type'        => Types::id(),
+					'description' => __( 'The ID of the deleted mediaItem', 'wp-graphql' ),
+					'resolve'     => function( $payload ) use ( $post_type_object ) {
+						$deleted = (object) $payload['mediaItemObject'];
+						return ! empty( $deleted->ID ) ? Relay::toGlobalId( $post_type_object->name, absint( $deleted->ID ) ) : null;
+					},
+				],
+				'mediaItem' => [
+					'type'        => Types::post_object( $post_type_object->name ),
+					'description' => __( 'The mediaItem before it was deleted', 'wp-graphql' ),
+					'resolve'     => function( $payload ) {
+						$deleted = (object) $payload['mediaItemObject'];
+						return ! empty( $deleted ) ? $deleted : null;
+					},
+				],
+			],
+			'mutateAndGetPayload' => function( $input ) use ( $post_type_object, $mutation_name ) {
+
+				/**
+				 * Get the ID from the global ID
+				 */
+				$id_parts = Relay::fromGlobalId( $input['id'] );
+				$existing_media_item = get_post( absint( $id_parts['id'] ) );
+
+				/**
+				 * If there's no existing mediaItem, throw an exception
+				 */
+				if ( empty( $existing_media_item ) ) {
+					throw new \Exception( __( 'No mediaItem could be found to delete', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Stop now if a user isn't allowed to delete a mediaItem
+				 */
+				if ( ! current_user_can( $post_type_object->cap->delete_post, absint( $id_parts['id'] ) ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to delete mediaItems', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Check if we should force delete or not
+				 */
+				$force_delete = ( ! empty( $input['forceDelete'] ) && true === $input['forceDelete'] ) ? true : false;
+
+				/**
+				 * Get the mediaItem object before deleting it
+				 */
+				$media_item_before_delete = get_post( absint( $id_parts['id'] ) );
+
+				/**
+				 * If the mediaItem is already in the trash, and the forceDelete input was not passed,
+				 * don't remove from the trash
+				 */
+				if ( 'trash' === $media_item_before_delete->post_status ) {
+					if ( true !== $force_delete ) {
+						// Translators: the first placeholder is the post_type of the object being deleted and the second placeholder is the unique ID of that object
+						throw new \Exception( sprintf( __( 'The mediaItem with id %1$s is already in the trash. To remove from the trash, use the forceDelete input', 'wp-graphql' ), $input['id'] ) );
+					}
+				}
+
+				/**
+				 * Delete the mediaItem
+				 */
+				$deleted = wp_delete_attachment( $id_parts['id'], $force_delete );
```

Comment:
Since this can return `false` on failure, we should account for that and throw an exception